### ISSUE TITLE
Correlation and Covariance that operate on (float * float) seq ?

### ISSUE THREAD

**Issue by nhirschey** [fslaborg/FSharp.Stats#123]
**Is your feature request related to a problem? Please describe.**

This is relatively minor as things go, but I'm frequently finding myself in this situation:

```fsharp
let data =
  [ 1.0, 1.1
     0.0, 1.05]
```
I want a correlation of the first and second thing in the tuples. So I am doing

```fsharp
let x, y = data |> Seq.unzip 

Seq.pearson x y
Seq.cov x y
```

**Describe the solution you'd like**
I think it would be more natural to be able to do something like

```fsharp
data |> Seq.pearson
```
I also think that it is easier to be confident that the `x` and `y` values are paired correctly with this syntax because you haven't had to deconstruct things. 

I agree that the current functionality of `Seq.pearson x y` is needed, but I thought I'd put it out there that this alternate version would also be useful.

**Describe alternatives you've considered**

**Additional context**

**Comment by nhirschey** [comment]
maybe Seq.pearsonOfPairs?

**Comment by bvenn** [comment]
While reading the issue I noticed that Seq.pearson definetely lacks in a check if both sequences are the same length. 

I like your suggestion of a ...OfPairs naming style 👍

**Comment by nhirschey** [comment]
If you'd like, I could work on a pull request implementing Seq.pearsonOfPairs, etc. and the length check.

I could use the length checking code in Seq.cov: https://github.com/fslaborg/FSharp.Stats/blob/28af6ee76eda67f27b3d86eb84c79321bc4ecc81/src/FSharp.Stats/Seq.fs#L638

**Comment by bvenn** [comment]
Sounds great! Thank you 👍

**Comment by bvenn** [comment]
closed by 4a07244
